### PR TITLE
Improving stdlib

### PR DESCRIPTION
**Pull request by mahyarsamani**

### CODE REVIEW

**Comment by powerjg** [comment]
Can you add some details to this PR in the description on how/why you are "improving the stdlib"?

Please give the reviewers some context so they can provide high quality reviews.

File: src/python/gem5/components/cachehierarchies/chi/nodes/dma_requestor.py [hunk 1]
```diff
@@ -58,8 +58,8 @@ def __init__(self, network, cache_line_size, clk_domain: ClockDomain):
         self.alloc_on_writeback = False
         self.dealloc_on_unique = False
         self.dealloc_on_shared = False
-        self.dealloc_backinv_unique = True
-        self.dealloc_backinv_shared = True
+        self.dealloc_backinv_unique = False
```
Comment (powerjg) [inline src/python/gem5/components/cachehierarchies/chi/nodes/dma_requestor.py:61]: Is this a bug fix?

File: src/python/gem5/components/memory/memory.py [hunk 2]
```diff
@@ -170,6 +170,10 @@ def get_mem_ports(self) -> Sequence[Tuple[AddrRange, Port]]:
     def get_memory_controllers(self) -> List[MemCtrl]:
         return [ctrl for ctrl in self.mem_ctrl]
 
+    @overrides(AbstractMemorySystem)
+    def get_monolithic_range(self) -> int:
```
Comment (powerjg) [inline src/python/gem5/components/memory/memory.py:174]: How important is this change?

**Review by powerjg — COMMENTED** [review]
We need some updates on this change *ASAP* since the staging branch is being creating on Friday

**Comment by BobbyRBruce** [comment]
I've cherry-picked the change I see as being of-value here to #665. I'm closing this PR as it appears to have been abandoned.

**Pull request closed without merging**